include/driver/target.h

```diff
@@ -40,26 +45,41 @@ class CPU : public Target {
 };
 
 class GPU : public Target {
-    Opt<std::pair<int, int>> computeCapability_;
+#ifdef FT_WITH_CUDA
+    Ref<cudaDeviceProp> infoArch_;
+#endif // FT_WITH_CUDA
 
   public:
+#ifndef FT_WITH_CUDA
     GPU(bool useNativeArch = true) : Target(useNativeArch) {}
+#endif // NOT FT_WITH_CUDA
```

Comment:
You can just enclose the whole `GPU` class with `#ifdef FT_WITH_CUDA`. If some tests fail, mark the test as requiring CUDA.